### ISSUE TITLE
GraphQL component nested filtering on booleans not working

### ISSUE THREAD

**Issue by EmilyYond** [strapi/strapi#17421]
### Required System information

- Node.js version: 18.16.1
- NPM version: 9.7.2
- Strapi version: 4.11.2
- Database: postgres
- Is your project Javascript or Typescript: Js

### Describe the bug

I am trying to separate a component called 'bookables' to return as two arrays depending on whether the 'featured' field (boolean) is true or false. But both queries just return all of the bookables, neither are being filtered. 

If I change the filter to fitler by any other field (eg. title, a string field) then it works and filters correctly. Is there a bug with filtering by booleans?

### Steps to reproduce the behavior

Try to filter by a boolean on a child component

### Expected behavior

The query should return two arrays, one of the 'featured bookables' and one of the 'unfeatured bookables'.

### Screenshots

![Screenshot_2023-07-21_at_09 57 03](https://github.com/strapi/strapi/assets/128067060/52180cbd-d918-4a5e-9b08-b21da79b4c2f)

**Comment by EmilyYond** [comment]
I think this is also related to https://forum.strapi.io/t/graphql-query-filtering-children-boolean-value/24110 which is the only thing I can find thats relevant, but no one replied to them with a solution.

**Comment by derrickmehaffy** [comment]
Can't reproduce on v4.11.7:

![image](https://github.com/strapi/strapi/assets/8593673/e323616b-2a6b-4f0a-84f1-0d81595c986d)

![image](https://github.com/strapi/strapi/assets/8593673/a3b02e32-648e-409c-88ec-5f794f892e7e)

**Comment by EmilyYond** [comment]
Hello @derrickmehaffy, thank you for looking into this. I'm not sure why yours is working, when I create a new strapi and replicate it I have the same issue again (see screenshot). 

![Screenshot 2023-07-25 at 09 53 57](https://github.com/strapi/strapi/assets/128067060/18d2f55a-f277-4bad-9ef9-2492a6985ff4)

I have attached my testapp project:
[testapp.zip](https://github.com/strapi/strapi/files/12159067/testapp.zip)

**Comment by derrickmehaffy** [comment]
> Hello @derrickmehaffy, thank you for looking into this. I'm not sure why yours is working, when I create a new strapi and replicate it I have the same issue again (see screenshot).
> 
> ![Screenshot 2023-07-25 at 09 53 57](https://user-images.githubusercontent.com/128067060/255846028-18d2f55a-f277-4bad-9ef9-2492a6985ff4.png)
> 
> I have attached my testapp project: [testapp.zip](https://github.com/strapi/strapi/files/12159067/testapp.zip)
> 
> I also upgraded my version from 4.11.2 to 4.11.7 but this didn't make a difference

I used your test app and still cannot reproduce it:
![image](https://github.com/strapi/strapi/assets/8593673/d26765d4-a540-4377-b849-7a3a00588b0b)
![image](https://github.com/strapi/strapi/assets/8593673/20fd0338-b585-4a72-920e-d3aeb135bec9)

**Comment by EmilyYond** [comment]
@derrickmehaffy yes exactly, its only when filtering nested components that it doesn't work. Thanks for continuing to investigate for me

**Comment by EmilyYond** [comment]
@derrickmehaffy do you have any update on this?
